### PR TITLE
MOVE-239 - Track more metadata about granting/revoking permissions to MVCR_READ role

### PR DESCRIPTION
**Pull request by ryanpdaley**
# Issue Addressed
This PR closes [MOVE-239](https://move-toronto.atlassian.net/browse/MOVE-239)

# Description
This adds four columns to the users table, and populates it when an admin gives or revokes a user's MVCR_READ permissions.

I’ve added:

- `mvcrAddedAt`: When (DateTime) the role is granted
- `mvcrRemovedAt`: When (DateTime) the role is revoked
- `mvcrAddedBy`: Who (user id) granted the role
- `mvcrRemovedB`: Who (user id) revoked the role

For clarity of the data, adding permissions to a user who had them revoked previously will clear out the previous revocation information. Ex:

- userA has the MVCR_READ permission. It was given by adminA on DateA. The JSON representation of the database entry for userA will look like:

```
{
  "id":3,
  "createdAt":"2024-04-11 16:56:45.120",
  "email":"tops01@o365.sandbox-toronto.ca",
  "sub":"23iZ6yetpZiubK7IbaD0+TATyLJL3RDLJ6CsjvdXUho=",
  "uniqueName":"CHILD\\tops01",
  "scope":["STUDY_REQUESTS","MVCR_READ"],
  "mvcrAddedAt":"2022-04-12 13:13:51.152",
  "mvcrRemovedAt":null,
  "mvcrAddedBy":2,
  "mvcrRemovedBy":null
}
```

- On DateB, adminB revokes the MVCR_READ permission of userA. The JSON representation of the database entry for userA will look like:

```
{
  "id":3,
  "createdAt":"2024-04-11 16:56:45.120",
  "email":"tops01@o365.sandbox-toronto.ca",
  "sub":"23iZ6yetpZiubK7IbaD0+TATyLJL3RDLJ6CsjvdXUho=",
  "uniqueName":"CHILD\\tops01",
  "scope":["STUDY_REQUESTS"],
  "mvcrAddedAt":"2022-04-12 13:13:51.152",
  "mvcrRemovedAt":"2023-05-12 09:15:57.340",
  "mvcrAddedBy":2,
  "mvcrRemovedBy":4
}
```
- On DateC, adminC adds the MVCR_READ permission of userA. The JSON representation of the database entry for userA will look like:
```
{
  "id":3,
  "createdAt":"2024-04-11 16:56:45.120",
  "email":"tops01@o365.sandbox-toronto.ca",
  "sub":"23iZ6yetpZiubK7IbaD0+TATyLJL3RDLJ6CsjvdXUho=",
  "uniqueName":"CHILD\\tops01",
  "scope":["STUDY_REQUESTS"],
  "mvcrAddedAt":"2024-04-12 09:13:51.152",
  "mvcrRemovedAt":null,
  "mvcrAddedBy":2,
  "mvcrRemovedBy":null
}
```
# Tests
All existing test are passing, and I have successfully added and revoked the MVCR_READ permissions of users with various other permissions. I have also added new users, and tested that one admin can add the permission, and another can remove it...

### CODE REVIEW

File: web/components/admin/FcAdminPermissions.vue [hunk 1]
```diff
@@ -18,7 +18,7 @@
             sort-by="UNIQUE_NAME"
             :sort-desc="false"
             :sort-keys="sortKeys">
-            <template v-slot:item.UNIQUE_NAME="{ item }">
+            <template v-slot:[`item.UNIQUE_NAME`]="{ item }">
```
Comment (ryanpdaley) [inline web/components/admin/FcAdminPermissions.vue:21]: This was a linter complaint. Has nothing to do with this PR, but I needed to fix it to commit. 🤷

File: web/components/admin/FcAdminPermissions.vue [hunk 2]
```diff
@@ -98,9 +98,9 @@ export default {
         value: 'UNIQUE_NAME',
         text: 'User',
       },
-      ...AuthScope.enumValues.map(({ name }) => ({
+      ...AuthScope.enumValues.map(({ name, description }) => ({
```
Comment (ryanpdaley) [inline web/components/admin/FcAdminPermissions.vue:101]: This was scope creep... we were using the name (MVCR_READ) rather than the description (View and Download MVCRs) in the table headers... so I changed it.

**Review by spencermountain — APPROVED** [review]
nice work!

**Review by lowsound42 — APPROVED** [review]
